parsec/backend/memory/block.py

```diff
@@ -35,28 +36,29 @@ def register_components(
         self._realm_component = realm
 
     def _check_realm_read_access(self, organization_id, realm_id, user_id):
-        can_read_roles = (
-            RealmRole.OWNER,
-            RealmRole.MANAGER,
-            RealmRole.CONTRIBUTOR,
-            RealmRole.READER,
-        )
-        self._check_realm_access(organization_id, realm_id, user_id, can_read_roles)
+        self._check_realm_access(organization_id, realm_id, user_id, OperationKind.DATA_READ)
 
     def _check_realm_write_access(self, organization_id, realm_id, user_id):
-        can_write_roles = (RealmRole.OWNER, RealmRole.MANAGER, RealmRole.CONTRIBUTOR)
-        self._check_realm_access(organization_id, realm_id, user_id, can_write_roles)
+        self._check_realm_access(organization_id, realm_id, user_id, OperationKind.DATA_WRITE)
 
-    def _check_realm_access(self, organization_id, realm_id, user_id, allowed_roles):
+    def _check_realm_access(self, organization_id, realm_id, user_id, operation_kind):
         try:
             realm = self._realm_component._get_realm(organization_id, realm_id)
         except RealmNotFoundError:
             raise BlockNotFoundError(f"Realm `{realm_id}` doesn't exist")
 
+        allowed_roles = (RealmRole.OWNER, RealmRole.MANAGER, RealmRole.CONTRIBUTOR)
```

Comment:
I would replace this by multiple if
```python
if operation_kind == OperationKind.DATA_READ:
    allowed_roles = (...)
elif operation_kind == OperationKind.DATA_WRITE:
    allowed_roles = (...)
elif operation_kind == OperationKind.MAINTENANCE:
    allowed_roles = (...)
else:
    allowed_roles = ()
```
This avoid the combination trick which grow increasingly unclear with the number of possible operation_kind

Reply:
on top of that, this is what is done in `memory/vlob.py` ^^

Reply:
9ddb38c5be108a3fc89d6ff4ab2404d92401c0ed